### PR TITLE
Disable Email Notifications

### PR DESCRIPTION
**Pull request by kobergj**
Adds new setting to disable email  notifications

### CODE REVIEW

**Comment by ownclouders** [comment]
E2E tests failed: https://drone.owncloud.com/owncloud/ocis/21745/68/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/21745/tracing/space-members-can-be-invited-with-an-expiration-date-alice-2023-4-25-10-57-58.zip 


</p></details>

**Comment by mmattel** [comment]
Needs webui documentation.
Will appear as item (when impemented by the web team) in the users personal settings.
Same location where the GDPR report request.

**Review by 2403905 — APPROVED** [review]

**Review by micbar — APPROVED** [review]

File: services/settings/pkg/store/defaults/defaults.go [hunk 1]
```diff
@@ -603,6 +672,16 @@ func generateBundleProfileRequest() *settingsmsg.Bundle {
 				},
 				Value: &languageSetting,
 			},
+			{
+				Id:          SettingUUIDProfileDisableNotifications,
+				Name:        "disable email notifications",
```
Comment (kulmann) [inline services/settings/pkg/store/defaults/defaults.go:677]: Possible to use `disable-email-notifications` here? I'd prefer to keep blanks out of identifiers that are referenced in code. The web ui uses this name to identify the setting.

File: services/settings/pkg/store/defaults/defaults.go [hunk 2]
```diff
@@ -33,6 +33,13 @@ const (
 	// LanguageReadWriteName is the hardcoded setting name for the language read write permission
 	LanguageReadWriteName string = "language-readwrite"
 
+	// DisableEmailNotificationsPermissionID is the hardcoded setting UUID for the language read write permission
+	DisableEmailNotificationsPermissionID string = "ad5bb5e5-dc13-4cd3-9304-09a424564ea8"
+	// DisableEmailNotificationsPermissionName is the hardcoded setting name for the language read write permission
+	DisableEmailNotificationsPermissionName string = "disable-email-notifications"
```
Comment (kulmann) [inline services/settings/pkg/store/defaults/defaults.go:39]: If I recall correctly there was some decision to go with permission names like `EmailNotifications.Disable` in the future (first one was `PublicLink.Write`). Don't know the details of the decision and how to form permission names then, but this caught my eye...

**Review by kulmann — CHANGES_REQUESTED** [review]
Would love to see my two comments being discussed. Putting a `request changes` here to prevent accidental merges. If my comments are bogus please dismiss my review.

File: services/settings/pkg/store/defaults/defaults.go [hunk 3]
```diff
@@ -33,6 +33,13 @@ const (
 	// LanguageReadWriteName is the hardcoded setting name for the language read write permission
 	LanguageReadWriteName string = "language-readwrite"
 
+	// DisableEmailNotificationsPermissionID is the hardcoded setting UUID for the language read write permission
```
Comment (JanAckermann) [inline services/settings/pkg/store/defaults/defaults.go:36]: The description is wrong -  not language

File: services/settings/pkg/store/defaults/defaults.go [hunk 4]
```diff
@@ -33,6 +33,13 @@ const (
...
+	DisableEmailNotificationsPermissionID string = "ad5bb5e5-dc13-4cd3-9304-09a424564ea8"
+	// DisableEmailNotificationsPermissionName is the hardcoded setting name for the language read write permission
```
Comment (JanAckermann) [inline services/settings/pkg/store/defaults/defaults.go:38]: Same here ^

**Review by JanAckermann — CHANGES_REQUESTED** [review]

**Review by JanAckermann — APPROVED** [review]

**Pull request merged**